sha1/src/lib.rs

```diff
@@ -62,38 +70,77 @@ impl UpdateCore for Sha1Core {
     fn update_blocks(&mut self, blocks: &[Block<Self>]) {
         self.block_len += blocks.len() as u64;
         let blocks = Array::cast_slice_to_core(blocks);
-        compress(&mut self.h, blocks);
+
+        #[cfg(feature = "collision")]
+        {
+            if let Some(ref mut ctx) = self.detection {
```

Comment:
I am not sure we should modify the "original" SHA-1 implementation in such fashion. We probably should keep it vulnerable and strongly advice to use the hardened version if possible in the crate docs.

Reply:
fixed, it is now fully standalone